app/models/message.rb

```diff
@@ -2,13 +2,20 @@ class Message < ActiveRecord::Base
   belongs_to :user
 
   validates_presence_of :to_phone_number, :body
+  validate :user_quota_limit
+
 
   after_commit :send_sms_job
 
   attr_reader :group_id
 
+  def user_quota_limit
+    self.user.quota_limit_check
+    errors.add(:message, "would exceed your quota. So it cannot be sent.")
```

Comment:
Ah, nice. I should have checked the code in master. I've got one more comment :)

As an reader of this code, it's sort of difficult to tell what the quota_limit_check method does. I had to read through the method's body to figure out. It might be more clear to rename the method to something like "exceeded_quota_limit?" (of course then you'd have to update the method body and change the 'unless' to an 'if'). Adding the question mark tells me that the method returns a boolean - a nice convention used by lots of rubyists as well as the standard library. Additionally, using the word "exceeded" rather than "check" is less ambiguous in this context, imo.